### PR TITLE
feat: add firebase core extensions

### PR DESCRIPTION
**Pull request by qu0cquyen**
## Status

Done

## Description
Add Firebase core extensions 
## Type of Change

- [X] ✨ New feature (non-breaking change which adds functionality)

### CODE REVIEW

**Comment by vuongxuongminh** [comment]
Hi @qu0cquyen, should we rename the package to `robusta_firebase_core`? Because all `Firebase` packages only support `Flutter` SDK and them not have the prefix or suffix `flutter`.

File: packages/flutter_robusta_firebase_core/example/analysis_options.yaml [hunk 1]
```diff
@@ -0,0 +1,4 @@
+include: package:very_good_analysis/analysis_options.3.1.0.yaml
```
Comment (vuongxuongminh) [inline packages/flutter_robusta_firebase_core/example/analysis_options.yaml]: Should use the latest version `4.0.0` and maybe you need to upgrade `very_good_cli` to generate up-to-date template.

File: packages/flutter_robusta_firebase_core/lib/src/extension.dart [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+  const FlutterFirebaseCoreExtension({String? name, FirebaseOptions? options})
+      : _name = name,
+        _firebaseOptions = options;
+
+  final String? _name;
+  final FirebaseOptions? _firebaseOptions;
+
+  @override
+  void load(Configurator configurator) {
+    configurator.addBoot(_boot, priority: 5429);
```
Comment (vuongxuongminh) [inline packages/flutter_robusta_firebase_core/lib/src/extension.dart]: The priority should be `2^n`, n should be lower than 8192 (priority of `FlutterAppExtension`).

File: packages/flutter_robusta_firebase_core/test/src/flutter_robusta_firebase_core_test.dart [hunk 3]
```diff
@@ -0,0 +1,42 @@
...
+    // test('able to use firebase app after providing extension', () async {
+    //   final runner = Runner(extensions: [
+    //     FlutterFirebaseCoreExtension(),
+    //     EventExtension<RunEvent>({
+    //       (e) => expect(Firebase.app(), returnsNormally): 0,
+    //     })
+    //   ]);
+
+    //   await runner.run();
+    // });
```
Comment (vuongxuongminh) [inline packages/flutter_robusta_firebase_core/test/src/flutter_robusta_firebase_core_test.dart]: Should remove this stub?

**Review by vuongxuongminh — CHANGES_REQUESTED** [review]

**Comment by qu0cquyen** [comment]
> Hi @qu0cquyen, should we rename the package to `robusta_firebase_core`? Because all `Firebase` packages only support `Flutter` SDK and them not have the prefix or suffix `flutter`.

If this is the case, then we will need to be consistent in all packages - in terms of packages naming. I.e: `flutter_robusta_hive` needs to change to `robusta_hive`. What do you think and please correct me if I'm wrong.

File: packages/flutter_robusta_firebase_core/lib/src/extension.dart (hunk 2 above)
Comment (qu0cquyen) [reply packages/flutter_robusta_firebase_core/lib/src/extension.dart]: Could you elaborate more about this for my future references? I'm gonna change it to `8192` as of now.

File: packages/flutter_robusta_firebase_core/test/src/flutter_robusta_firebase_core_test.dart (hunk 3 above)
Comment (qu0cquyen) [reply packages/flutter_robusta_firebase_core/test/src/flutter_robusta_firebase_core_test.dart]: Done.

File: packages/robusta_firebase_core/lib/robusta_firebase_core.dart [hunk 4]
```diff
@@ -0,0 +1,4 @@
+/// A Very Good Project created by Very Good CLI.
```
Comment (vuongxuongminh) [inline packages/robusta_firebase_core/lib/robusta_firebase_core.dart]: Please add package description.

File: packages/robusta_firebase_core/README.md [hunk 5]
```diff
@@ -0,0 +1,33 @@
+# Flutter Robusta Firebase Core
```
Comment (vuongxuongminh) [inline packages/robusta_firebase_core/README.md]: Should be `Robusta Firebase Core`?

File: packages/robusta_firebase_core/README.md [hunk 6]
```diff
@@ -0,0 +1,33 @@
...
+# Prerequisites:
+
+Please refer to FlutterFire docs for init configuration (https://firebase.flutter.dev/docs/overview)
+
+## Installation
+
+Install this package via pub command:
+
+```
+flutter pub add flutter_robusta_firebase_core
```
Comment (vuongxuongminh) [inline packages/robusta_firebase_core/README.md]: Should be `robusta_firebase_core`?

File: packages/robusta_firebase_core/README.md [hunk 7]
```diff
@@ -0,0 +1,33 @@
...
+```
+
+## Usages
+
+Add `FlutterFirebaseCoreExtension` to runner:
+
+```dart
+import 'package:flutter_robusta/flutter_robusta.dart';
+import 'package:flutter_robusta_firebase_core/flutter_robusta_firebase_core.dart';
```
Comment (vuongxuongminh) [inline packages/robusta_firebase_core/README.md]: Like concerns above

**Review by vuongxuongminh — CHANGES_REQUESTED** [review]

**Review by vuongxuongminh — APPROVED** [review]

**Comment by vuongxuongminh** [comment]
Merged, thank you @qu0cquyen

**Pull request merged**